src/EditorFeatures/CSharpTest/Diagnostics/PopulateSwitch/PopulateSwitchTests.cs

```diff
@@ -0,0 +1,877 @@
+﻿using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.CodeFixes.PopulateSwitch;
+using Microsoft.CodeAnalysis.CSharp.Diagnostics.PopulateSwitch;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Test.Utilities;
+using System;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics.PopulateSwitch
+{
+    public partial class PopulateSwitchTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override Tuple<DiagnosticAnalyzer, CodeFixProvider> CreateDiagnosticProviderAndFixer(Workspace workspace)
+        {
+            return new Tuple<DiagnosticAnalyzer, CodeFixProvider>(
+                new CSharpPopulateSwitchDiagnosticAnalyzer(), new PopulateSwitchCodeFixProvider());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsPopulateSwitch)]
+        public async Task AllMembersAndDefaultExist()
+        {
+            await TestMissingAsync(
+            @"
+namespace ConsoleApplication1
+{
+    enum MyEnum
+    {
+        Fizz, Buzz, FizzBuzz
+    }
+    class MyClass
+    {
+        void Method()
+        {
+            var e = MyEnum.Fizz;
+            switch (e)
+            {
+                [|case MyEnum.Fizz:
+                case MyEnum.Buzz:
+                case MyEnum.FizzBuzz:
+                default:
+                    break;|]
+            }
+        }
+    }
+}
+");
```

Comment:
Yes, the fix is provided when my cursor is in the `switch` keyword.  I put the span around the case statements in the switch because that is what is changed.  Should I move them to be around the entire switch/select block?